### PR TITLE
Add improvement to SCIM metadata generation for hybrid roles

### PR DESCRIPTION
**Pull request by sachiniWettasinghe**
Fixes wso2/product-is#5702

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java [hunk 1]
```diff
@@ -174,6 +174,41 @@ public void addSCIMGroupAttributes(int tenantId, String roleName, Map<String, St
...
+                    } else {
+                        throw new IdentitySCIMException("Error when adding SCIM Attribute: " + entry.getKey() +
+                                ". An attribute with the same name already exists.");
+                    }
+                }
+            }
+            prepStmt.executeBatch();
+            connection.commit();
+        } catch (SQLException e) {
+            throw new IdentitySCIMException("Error when adding SCIM attributes for groups.", e);
```
Comment (denuwanthi) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: Shall we mention hybrid groups?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -3046,6 +3062,22 @@ private boolean isDomainSupportedAttribute(String attributeName) {
         }
     }
 
+    private String[] checkForSCIMDisabledHybridRoles(String[] roles) throws CharonException {
```
Comment (denuwanthi) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Add a method description

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -3046,6 +3062,22 @@ private boolean isDomainSupportedAttribute(String attributeName) {
...
+            Set<String> scimRoles = groupHandler.listSCIMRoles();
+            List<String> scimDisabledHybridRoles = getSCIMDisabledHybridRoleList(roles, scimRoles);
+            if (!scimDisabledHybridRoles.isEmpty()) {
+                createSCIMAttributesForSCIMDisabledHybridRoles(scimDisabledHybridRoles);
+            }
+        } catch (org.wso2.carbon.user.core.UserStoreException | IdentitySCIMException e) {
+            throw new CharonException("Error in retrieving SCIM Group information from database.", e);
+        }
+
+        return roles;
```
Comment (denuwanthi) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: What is the purpose of returning the roles as no modification is done to 'roles' within this method.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 4]
```diff
@@ -1885,7 +1888,14 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
 
         SCIMGroupHandler groupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
         if (StringUtils.isEmpty(domainName)) {
-            return groupHandler.listSCIMRoles();
+            Set<String> roleNames = groupHandler.listSCIMRoles();
```
Comment (denuwanthi) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: lets do the same logic in else section

Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 3 above)
Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Fixed

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 2 above)
Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: added

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java (hunk 1 above)
Comment (sachiniWettasinghe) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java]: fixed

**Pull request merged**